### ISSUE TITLE
Adding group owner as member

### ISSUE THREAD

**Issue by gr455** [CircuitVerse/CircuitVerse#2092]
**Describe the bug**
Currently it is possible to add the group mentor (owner) in the group as a member through mail and invite link, this is unnecessary.

**To Reproduce**
1. Create a Group
2. Add yourself to the group by clicking `Add Members` through email or the invite link

**Expected behavior**
Should not add owner to the group as a member

**Screenshots**
![Screenshot from 2021-01-27 21-44-38](https://user-images.githubusercontent.com/53974118/106019747-0015b080-60e9-11eb-8fb1-caac4fece623.png)

**Are you working on this issue? (Yes/No)**
No

**Comment by raghavendrashekhawat2** [comment]
Hi. I'd Like to work on this issue. As i see it if anyone tries to add themselves to a group they should get a flash saying that they can not be added to the group as they are the group owner ?.

**Comment by gr455** [comment]
No, that is not the problem. The problem is that if the mentor adds themself to the group, they get added in as a member too (note that this doesn't drop their mentor privileges), but this is redundant as a mentor is always in their group. So maybe flash a message saying 'Member already exists as the group mentor'

**Comment by raghavendrashekhawat2** [comment]
Yep cool. I am working on it

**Comment by raghavendrashekhawat2** [comment]
> No, that is not the problem. The problem is that if the mentor adds themself to the group, they get added in as a member too (note that this doesn't drop their mentor privileges), but this is redundant as a mentor is always in their group. So maybe flash a message saying 'Member already exists as the group mentor'

Whenever the mentor adds a user a message informing about the number of new users added, invited or if a user is already present is flashed. So instead of flashing something like "mentor already exists as a user" i am flashing "user is already present" does that work too ?

**Comment by gr455** [comment]
> > No, that is not the problem. The problem is that if the mentor adds themself to the group, they get added in as a member too (note that this doesn't drop their mentor privileges), but this is redundant as a mentor is always in their group. So maybe flash a message saying 'Member already exists as the group mentor'
> 
> Whenever the mentor adds a user a message informing about the number of new users added, invited or if a user is already present is flashed. So instead of flashing something like "mentor already exists as a user" i am flashing "user is already present" does that work too ?
> ![Capture](https://user-images.githubusercontent.com/70352627/107239251-abf6be80-6a4e-11eb-9d9a-eff4548fce27.JPG)

Hey! sorry for following up late. Yeah that looks good according to me!

**Comment by raghavendrashekhawat2** [comment]
Hi. No worries. I am new to open source, whenever i create a pull request on gitpod i get this error. I have given gitpod all the access but i still get this error. 
![image](https://user-images.githubusercontent.com/70352627/107848579-900e6680-6e1a-11eb-8468-e0386b88d2e6.png)

**Comment by gr455** [comment]
@raghavendrashekhawat2 that seems to be   a Bugnsag warning. This is expected since the API key is only set on the actual production server.

They does not affect the functioning of the app locally, so don't worry

**Comment by raghavendrashekhawat2** [comment]
@gr455  Hi. Sorry for continuing this a little late. In my pr @tachyons mentioned that my pr does not address the group invite link and the api. I have fixed the group invite link but i am having some trouble understanding the api could you help me with it?. Does api refer to the documentation here  https://api.circuitverse.org/#group-members ?. If yes what would i need to add in the documentation ?. Thanks!

**Comment by gr455** [comment]
@raghavendrashekhawat2, Yes that's the documentation for the API. In the documentation under https://api.circuitverse.org/#create-add-group-member, you might want to add another status code 409 saying the group member already exists. If you have any questions regarding the API you can ask those too

**Comment by raghavendrashekhawat2** [comment]
@gr455 Thanks !. I would need to add the error code in the api doc or would i need to return a 409 in the code too ?.

**Comment by gr455** [comment]
> @gr455 Thanks !. I would need to add the error code in the api doc only or would i need to return a 409 in the code too ?.

I think you will need to return it in the API too

**Comment by vedant-jain03** [comment]
@gr455 
Can I work on this?

**Comment by gr455** [comment]
> @gr455 Can I work on this?
Yeah go ahead. Make sure to make the changes on the API side as well
